toolchain/check/testdata/class/fail_generic_method.carbon

```diff
@@ -42,6 +42,7 @@ fn Class(N:! i32).F[self: Self](n: T) {}
 // CHECK:STDOUT:   %Class: type = class_type @Class, @Class(%T) [symbolic]
 // CHECK:STDOUT:   %require_complete.4ae: <witness> = require_complete_type %T [symbolic]
 // CHECK:STDOUT:   %Class.elem: type = unbound_element_type %Class, %T [symbolic]
+// CHECK:STDOUT:   %require_complete.78c: <witness> = require_complete_type %Class.elem [symbolic]
```

Comment:
Is it intended that we have additional require_complete_types happening here? We already have a `require_complete_type %T` above, which seems to be covering the same thing?

Reply:
I'm not very clear on this, but I think `unbound_element_type %Class, %T` is the type of the expression `Class.T`, so `require_complete_type %T` requires `T` itself to be complete, whereas `require_complete_type %Class.elem` requires the _type of_ `T` to be complete. I'm not clear on why `%Class.elem` isn't canonicalized to `type`, but given that it's being treated as a separate constant, it seems like it makes sense to require it to be complete. I don't know why it wasn't here before, though.

I'm asking on Discord to try to get more information about what's going on here.